main.js

```diff
@@ -0,0 +1,116 @@
+
+var towersOfHanoi = {
+
+  board: [
+    ['5', '4', '3', '2', '1'],
+    [],
+    []
+  ],
+  
+  moveDisc: function(startPeg, endPeg) {
+    //adjusts parameters so they can act on the correct index
+    startPeg -= 1;
+    endPeg -= 1; 
+    
+    var board = this.board; 
+    var start = board[startPeg]; //board[0] (peg 1) for move 1
+    var end = board[endPeg]; //board[1] (peg 3) for move 1
+  
+    //logic to moveDisc based on legal vs illegal moves
+    if(end.length === 0 || start[start.length-1] < end[end.length-1]) {
+
+      end.push(start.pop());
+      printBoard(board);
+
+      //checks for winner after each move
+      checkWinner(); 
+
+    } else {
+      console.log('That move is not allowed! Try again.');
+    };
+
+  }
+};
+
+//prints board to console horizontally
+var printBoard = function(b) {
+  var gameBoard = b.map(function (pegs) {
+    return console.log('--- ' + pegs.join(' '));  
+  });
+
+  console.log('_____________');
+  
+  return gameBoard; 
+};
+
+//checks winning condition and resets board if met
+var checkWinner = function() {
+  var board = towersOfHanoi['board'];
+  var win = board.find(function() {
+    return board[1].length === 5 || board[2].length === 5
+  });
+
+  if(win) {
+    console.log('You won! Play again?')
+    reset();
```

Comment:
This is the only place `reset` is being called (which is fine), but `reset` is for whatever reason returning `printBoard` which returns a mapping of the original board, but nothing is being done with it.